### PR TITLE
grid: container as a readonly property

### PR DESCRIPTION
**Pull request by isidorn**
To make ti easier to move the grid to another parent.
imho simplified some code, no need to pass around container as much as before.
Tests pass, smoke tests pass, I tried it out and looks decent

fyi @bpasero

### CODE REVIEW

File: src/vs/base/browser/ui/grid/gridview.ts [hunk 1]
```diff
@@ -483,8 +483,11 @@ export class GridView implements IDisposable {
 		return this.root.maximumHeight;
 	}
 
-	constructor(container: HTMLElement, options: IGridViewOptions = {}) {
-		this.element = append(container, $('.monaco-grid-view'));
+	readonly container: HTMLElement;
+
+	constructor(options: IGridViewOptions = {}) {
+		this.container = $('.');
+		this.element = append(this.container, $('.monaco-grid-view'));
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/grid/gridview.ts]: No need to save any container. Expose `element` as `readonly`, and then:

```
this.element = $('.monaco-grid-view');
```

File: src/vs/base/test/browser/ui/grid/grid.test.ts [hunk 2]
```diff
@@ -8,19 +8,16 @@ import { Direction, Grid, getRelativeLocation, Orientation, SerializableGrid, IS
...
+	grid.container.style.height = `${600}px`;
 
-	setup(function () {
-		container = document.createElement('div');
-		container.style.position = 'absolute';
-		container.style.width = `${800}px`;
-		container.style.height = `${600}px`;
-	});
+	return grid;
+}
```
Comment (joaomoreno) [inline src/vs/base/test/browser/ui/grid/grid.test.ts]: You should leave the old behaviour of having a fixed size container. Then, create a grid and append it to the container. You probably still need `setup` and `teardown`.

File: src/vs/base/test/browser/ui/grid/grid.test.ts [hunk 3]
```diff
@@ -459,23 +465,19 @@ function nodesToNames(node: GridNode<TestSerializableView>): any {
...
+	const grid = new SerializableGrid(view);
+	grid.container.style.position = 'absolute';
+	grid.container.style.width = `${800}px`;
+	grid.container.style.height = `${600}px`;
 
-	teardown(function () {
-		container = null;
-	});
+	return grid;
+}
```
Comment (joaomoreno) [inline src/vs/base/test/browser/ui/grid/grid.test.ts]: Same as above.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Review by bpasero — APPROVED** [review]
I only checked the changes in `editorPart.ts`

**Pull request merged**